File: golang/67_add_binary.go
package main

import (
	"strconv"
)

func addBinary(a string, b string) string {
	carry, res, i := 0, "", len(a) - 1
	for  j := len(b)-1; (i >= 0 || j >= 0 || carry > 0); j = j-1{
		if i >= 0{
			carry += int(a[i] - '0')
		}
		if j >= 0{
			carry += int(b[j] - '0')
		}
		res, carry = strconv.Itoa(int(carry % 2)) + res, carry / 2
		i--
	}
	return res
}

File: leetcode/archived/Python/67_add_binary.py
#only works on int, will exceed this limit on other progmramming language

class Solution:
    def addBinary(self, a: str, b: str) -> str:
        def converttonumber(a):
            res, i = 0, 0
            while i < len(a):
                res = res * 2 + int(a[i])
                i += 1
            return res

        def convertnumbertostr(a):
            if a == 0:
                return "0"
            res = ""
            while a > 0:
                res += str(a % 2)
                a = a // 2
            return res[::-1]

        n1, n2 = converttonumber(a), converttonumber(b)
        return convertnumbertostr(n1 + n2)
